include/cocaine/detail/gateway/adhoc.hpp

```diff
@@ -47,12 +47,16 @@ class adhoc_t:
 
     // TODO: Make sure that remote service metadata is consistent across the whole cluster.
     synchronized<remote_map_t> m_remotes;
-    std::string local_uuid;
 
 public:
     adhoc_t(context_t& context, const std::string& _local_uuid, const std::string& name, const dynamic_t& args);
 
     auto
+    resolve_policy() const ->resolve_policy_t override {
+        return resolve_policy_t::remote_only;
```

Comment:
because it postulates so by it's policy. Locator tries to resolve local first in this case.